algoliasearch-common/src/main/java/com/algolia/search/AsyncObjects.java

```diff
@@ -200,6 +200,80 @@
     return getApiClient().saveObjects(getName(), objects, requestOptions);
   }
 
+  /**
+   * Replace all objects
+   *
+   * @param objects the list objects to update
+   * @param safe run the method in a safe way
+   */
+  default void replaceAllObjects(@Nonnull Iterable<T> objects, boolean safe)
+      throws ExecutionException, InterruptedException {
+    replaceAllObjects(objects, safe, new RequestOptions());
+  }
+
+  /**
+   * Replace all objects
+   *
+   * @param objects the list objects to update
+   * @param safe run the method in a safe way
+   * @param requestOptions Options to pass to this request
+   */
+  default void replaceAllObjects(
+      @Nonnull Iterable<T> objects, boolean safe, @Nonnull RequestOptions requestOptions)
+      throws ExecutionException, InterruptedException {
+
+    String tmpName = this.getName() + "_tmp_" + UUID.randomUUID().toString();
+    List<String> scope = new ArrayList<>(Arrays.asList("rules", "settings", "synonyms"));
+    List<AsyncTaskSingleIndex> batchResponses = new ArrayList<>();
+
+    // Copy all index resources from production index
+    AsyncTask copyIndexResponse =
+        getApiClient().copyIndex(getName(), tmpName, scope, requestOptions).get();
+
+    if (safe) {
+      getApiClient().waitTask(copyIndexResponse);
+    }
+
+    // Send records (batched automatically)
+    ArrayList<T> records = new ArrayList<>();
+
+    for (T object : objects) {
+
+      if (records.size() == 1000) {
```

Comment:
Performance issue ( at least on PHP ): You may want to increase a counter instead of calculating the size of the records every object.

Reply:
@nunomaduro which size did you take? 10 000?

Reply:
I am taking 1000.